ibis/pyspark/tests/test_timecontext.py

```diff
@@ -9,7 +14,408 @@
 def test_table_with_timecontext(client):
     table = client.table('time_indexed_table')
     context = (pd.Timestamp('20170102'), pd.Timestamp('20170103'))
-    result = table.compile(timecontext=context).toPandas()
-    expected = table.compile().toPandas()
+    result = table.execute(timecontext=context)
+    expected = table.execute()
     expected = expected[expected.time.between(*context)]
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    ('ibis_window', 'spark_range'),
+    [
+        ([('trailing', 1)], (-3600, 0)),  # 1h back looking window
```

Comment:
Can we just use the "preceding" and "following" as parameters? I think it matches the window definition battle

e.g.
```
(-ibis.interval(hours=1), 0), (-3600, 0)
```

Reply:
Can you add a test for both forward and trailing window?
```
(-ibis.interval(hours=1), ibis.interval(hours=1)), (-3600, 3600)
```

Reply:
Good idea to make preceding and following as parameters, the code is refactored and it is cleaner

Reply:
I added a test to test both direction.